peer/brontide.go

```diff
@@ -2538,13 +2590,59 @@ func (p *Brontide) reenableActiveChannels() {
 	// channel is already active, the update won't be sent.
 	for _, chanPoint := range activePublicChans {
 		err := p.cfg.ChanStatusMgr.RequestEnable(chanPoint, false)
-		if err == netann.ErrEnableManuallyDisabledChan {
-			p.log.Debugf("Channel(%v) was manually disabled, ignoring "+
-				"automatic enable request", chanPoint)
-		} else if err != nil {
-			p.log.Errorf("Unable to enable channel %v: %v",
-				chanPoint, err)
+
+		switch {
+		// No error occurred, continue to request the next channel.
+		case err == nil:
+			continue
+
+		// Cannot auto enable a manually disabled channel so we do
+		// nothing but proceed to the next channel.
+		case errors.Is(err, netann.ErrEnableManuallyDisabledChan):
+			p.log.Debugf("Channel(%v) was manually disabled, "+
+				"ignoring automatic enable request", chanPoint)
+
+			continue
+
+		// If the channel is reported as inactive, we will give it
+		// another chance. When handling the request, ChanStatusManager
+		// will check whether the link is active or not. One of the
+		// conditions is whether the link as been marked as
+		// reestablished, which happens inside a goroutine(htlcManager)
+		// after the link is started. And we may get a false negative
+		// saying the link is not active because that goroutine hasn't
+		// reached the line to mark the reestablishment. Thus we give
+		// it a second chance to send the request.
+		//
+		// TODO(yy): make the link signaling back its current state to
+		// properly handle this issue.
+		case errors.Is(err, netann.ErrEnableInactiveChan):
+			p.log.Warnf("Channel(%v) cannot be enabled as " +
+				"ChanStatusManager reported inactive, retrying")
+
+			// We now wait for retryEnableInterval and retry the
+			// request.
+			select {
+			case <-time.After(retryEnableInterval):
+			case <-p.quit:
+				p.log.Debug("Exit enabling channels due to " +
+					"peer shutdown")
+			}
+
+			// Send the request again.
+			err = p.cfg.ChanStatusMgr.RequestEnable(
+				chanPoint, false,
+			)
+			if err == nil {
+				continue
```

Comment:
So we wait for `retryEnableInterval` only once before exiting and giving up on enabling the channel?

Reply:
5 seconds is a lot, but it should retry

Reply:
> giving up on enabling the channel?

Yeah we'd give up. The only reason we give it a second chance is because the startup order of our subsystems may cause this "flake", and we wait and try. 5s is a lot, only added for itest and I think 1s should be enough.